src/main/java/de/mediathekview/mserver/crawler/srf/parser/SrfFilmJsonDeserializer.java

```diff
@@ -0,0 +1,351 @@
+    film.setBeschreibung(chapterList.description);
+    film.setWebsite(buildWebsiteUrl(chapterList.id, episodeData.title, theme));
+    addUrls(videoUrls, film);
+    addSubtitle(chapterList.subtitleUrl, film);    
+
+    return Optional.of(film);
+  }
+
+  private static void addSubtitle(String aSubtitleUrl, Film aFilm) {
+    if (!aSubtitleUrl.isEmpty()) {
+      try {
+        aFilm.addSubtitle(new URL(aSubtitleUrl));
+      } catch (MalformedURLException ex) {
+        LOG.error(String.format("A subtitle URL \"%s\" isn't valid.", aSubtitleUrl), ex);
+      }
+    }
+  }
+  
+  private static void addUrls(Map<Resolution, String> aVideoUrls, Film aFilm) {
+    aVideoUrls.entrySet().forEach(urlEntry -> {
+      try {
+        aFilm.addUrl(urlEntry.getKey(), CrawlerTool.uriToFilmUrl(new URL(urlEntry.getValue())));
+      } catch (MalformedURLException ex) {
+        LOG.error(String.format("A found download URL \"%s\" isn't valid.", urlEntry.getValue()), ex);
+      }
+    });
+  }
+  
+  private static Optional<URL> buildWebsiteUrl(String aId, String aTitle, String aTheme) {
+    
+    String url = String.format(WEBSITE_URL,
+            replaceCharForUrl(aTheme), replaceCharForUrl(aTitle), aId);
+
+    try {
+      return Optional.of(new URL(url));
+    } catch (MalformedURLException ex) {
+      LOG.error(String.format("The website url \"%s\" isn't valid.", url), ex);
+    }
+    
+    return Optional.empty();
+  }
+  
+  private static String replaceCharForUrl(String aValue) {
+    return aValue.toLowerCase().replace(' ', '-').replace('.', '-').replace(',', '-').replace(":", "").replace("\"", "")
+            .replace("--", "-");
+  }
+
+  private static String parseShow(JsonObject aJsonObject) {
+    if (aJsonObject.has(ELEMENT_SHOW)) {
+      JsonElement showElement = aJsonObject.get(ELEMENT_SHOW);
+      if (!showElement.isJsonNull()) {
+        JsonObject showObject = showElement.getAsJsonObject();
+        if (showObject.has(ATTRIBUTE_TITLE)) {
+          return showObject.get(ATTRIBUTE_TITLE).getAsString();
+        }
+      }
+    }
+
+    return "";
+  }
+
+  private EpisodeData parseEpisode(JsonObject aJsonObject) {
+    EpisodeData result = new EpisodeData();
+
+    if (aJsonObject.has(ELEMENT_EPISODE)) {
+      JsonElement episodeElement = aJsonObject.get(ELEMENT_EPISODE);
+
+      if (!episodeElement.isJsonNull()) {
+        JsonObject episodeObject = episodeElement.getAsJsonObject();
+
+        if (episodeObject.has(ATTRIBUTE_TITLE)) {
+          result.title = episodeObject.get(ATTRIBUTE_TITLE).getAsString();
+        }
+
+        if (episodeObject.has(ATTRIBUTE_PUBLISHED_DATE)) {
+          String date = episodeObject.get(ATTRIBUTE_PUBLISHED_DATE).getAsString();
+          result.publishDate = LocalDateTime.parse(date, DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+        }
+      }
+    }
+
+    return result;
+  }
+
+  private ChapterListData parseChapterList(JsonObject aJsonObject) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 26 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)